apps/studio/components/interfaces/Auth/Hooks/EnterpriseHooksConfig.tsx

```diff
@@ -0,0 +1,198 @@
+    if (ui.selectedProjectRef) meta.functions.load()
+  }, [ui.selectedProjectRef])
+
+  const onSubmit = (values: any, { resetForm }: any) => {
+    const payload = { ...values }
+    updateAuthConfig(
+      { projectRef: projectRef!, config: payload },
+      {
+        onError: () => {
+          ui.setNotification({
+            category: 'error',
+            message: `Failed to update settings`,
+          })
+        },
+        onSuccess: () => {
+          ui.setNotification({
+            category: 'success',
+            message: `Successfully updated settings`,
+          })
+          resetForm({ values: values, initialValues: values })
+        },
+      }
+    )
+  }
+
+  if (isError) {
+    return (
+      <Alert_Shadcn_ variant="destructive">
+        <IconAlertCircle strokeWidth={2} />
+        <AlertTitle_Shadcn_>Failed to retrieve auth configuration</AlertTitle_Shadcn_>
+        <AlertDescription_Shadcn_>{authConfigError.message}</AlertDescription_Shadcn_>
+      </Alert_Shadcn_>
+    )
+  }
+
+  return (
+    <Form id={FORM_ID} initialValues={INITIAL_VALUES} onSubmit={onSubmit} validationSchema={schema}>
+      {({ handleReset, resetForm, values, initialValues, setFieldValue }: any) => {
+        const hasChanges = JSON.stringify(values) !== JSON.stringify(initialValues)
+
+        // Form is reset once remote data is loaded in store
+        useEffect(() => {
+          if (isSuccess) {
+            resetForm({ values: INITIAL_VALUES, initialValues: INITIAL_VALUES })
+          }
+        }, [isSuccess])
+
+        return (
+          <>
+            <FormHeader
+              title="Enterprise Hooks"
+              description="Advanced Auth hooks available only to Teams and Enterprise plan customers."
+            />
+            <FormPanel
+              disabled={true}
+              footer={
+                <div className="flex py-4 px-8">
+                  <FormActions
+                    form={FORM_ID}
+                    isSubmitting={isUpdatingConfig}
+                    hasChanges={hasChanges}
+                    handleReset={handleReset}
+                    disabled={!canUpdateConfig || !isTeamsEnterprisePlan}
+                    helper={
+                      !canUpdateConfig
+                        ? 'You need additional permissions to update authentication settings'
+                        : undefined
+                    }
+                  />
+                </div>
+              }
+            >
+              <FormSection header={<FormSectionLabel>MFA Verification Attempt</FormSectionLabel>}>
+                <FormSectionContent loading={isLoading}>
+                  <SchemaFunctionSelector
+                    id="HOOK_MFA_VERIFICATION_ATTEMPT_URI"
+                    descriptionText="Select the function to be called by Supabase Auth each time a user tries to verify an MFA factor. Return a decision on whether to reject the attempt and future ones, or to allow the user to keep trying."
```

Comment:
"Return a decision ..." could be reworded I think. Though I'm not familiar with what this does